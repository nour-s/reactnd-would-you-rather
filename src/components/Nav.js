import React from "react";
import { NavLink } from "react-router-dom";

export default class Nav extends React.Component {
	render() {
		return (
			<React.Fragment>
				<nav className="nav">
					<ul>
						<li>
							<NavLink to="/" exact activeClassName="active">
								Home
							</NavLink>
						</li>
						<li>
							<NavLink to="/add" exact activeClassName="active">
								New Question
							</NavLink>
						</li>
						<li>
							<NavLink
								to="/leaderboard"
								exact
								activeClassName="active"
							>
								Leader Board
							</NavLink>
						</li>
					</ul>
				</nav>
			</React.Fragment>
		);
	}
}
