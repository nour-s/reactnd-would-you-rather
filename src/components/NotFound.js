import React from "react";

const NotFound = () => (
	<div className="not-found">
		<h1>404 Poll not found</h1>
	</div>
);

export default NotFound;
