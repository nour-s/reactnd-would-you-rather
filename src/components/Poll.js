import React, { Component } from 'react'

class Poll extends Component {
  
  render(){
    const poll = this.props.poll || {};
    
    return (
      <div>
      <h1>
      {JSON.stringify(poll)}
      </h1>
            </div>
           );
  }
}

export default Poll;