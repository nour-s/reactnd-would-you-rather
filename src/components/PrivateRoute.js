import React from "react";
import { withRouter, Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";

const PrivateRoute = ({ component: Component, ...routeProps }) => {
	const { authedUser } = routeProps;
	// if render method was already passed, then use it, otherwise render the component
	const render = routeProps.render || (props => <Component {...props} />);
	const redirect = props => (
		<Redirect
			to={{ pathname: "/login", state: { from: props.location } }}
		/>
	);

	return (
		<Route
			{...routeProps}
			render={props => (authedUser.id ? render(props) : redirect(props))}
		/>
	);
};

const mapStateToProps = ({ authedUser }) => ({
	authedUser
});

export default withRouter(connect(mapStateToProps)(PrivateRoute));
