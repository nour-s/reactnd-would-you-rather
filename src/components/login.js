import React, { Component } from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import { bindActionCreators } from "redux";
import { setAuthedUser } from "../actions/shared";
import UserSummary from "../components/userSummary";
import "./login.scss";

class Login extends Component {
	state = {
		selectedUserId: "",
		showError: false,
		isAuthorized: false
	};

	handleSubmit = e => {
		e.preventDefault();
		const { selectedUserId } = this.state;
		if (!selectedUserId || selectedUserId === "") {
			this.setState({ showError: true });
			return;
		}
		this.props.setAuthedUser(this.state.selectedUserId);
		this.setState(prevState => ({ ...prevState, isAuthorized: true }));
	};

	handleChange = e => {
		this.setState({ selectedUserId: e.target.value, showError: false });
	};

	render() {
		const { from } = this.props.location.state || {
			from: { pathname: "/" }
		};

		const user = this.props.users[this.state.selectedUserId];

		if (this.state.isAuthorized) {
			return <Redirect to={from} />;
		}

		let users = Object.keys(this.props.users).map(
			uid => this.props.users[uid]
		);

		return (
			<div className="login">
				<h1 className="login_title">Login</h1>
				<UserSummary user={user} />
				<form onSubmit={this.handleSubmit}>
					{this.state.showError && (
						<span className="validation_error">
							You should select a user
						</span>
					)}
					<div className="login_select-wrapper">
						<select
							value={this.state.selectedUserId}
							onChange={this.handleChange}
							className="login_select"
						>
							<option key="empty" />
							{users.map(user => (
								<option key={user.id} value={user.id}>
									{user.name}
								</option>
							))}
						</select>
					</div>
					<button className="login_submit">Login</button>
				</form>
			</div>
		);
	}
}

const mapStateToProps = ({ authedUser, users }) => ({
	authedUser,
	users
});

const mapDispatcToProps = dispatch => {
	return bindActionCreators({ setAuthedUser }, dispatch);
};

export default connect(
	mapStateToProps,
	mapDispatcToProps
)(Login);
