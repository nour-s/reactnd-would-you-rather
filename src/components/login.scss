.validation_error {
	color: #f00;
}
