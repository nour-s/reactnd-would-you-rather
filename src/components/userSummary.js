import React from "react";
import PropTypes from "prop-types";

const UserSummary = ({
	user = { avatarURL: "../../images/default.png", name: "Select User" }
}) => (
	<div className="author-info">
		<img src={user.avatarURL} alt="avatar" />
		<span>{user.name}</span>
	</div>
);

UserSummary.propTypes = {
	user: PropTypes.object.isRequired
};

export default UserSummary;
