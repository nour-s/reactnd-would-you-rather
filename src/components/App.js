import React, { Fragment, Component } from "react";
import { connect } from "react-redux";
import { handleInitialData } from "../actions/shared";
import { BrowserRouter as Router, Route } from "react-router-dom";
// import LoadingBar from "react-redux-loading";
import PollList from "../components/Poll/pollList";
import Poll from "../components/Poll/poll";
import CreatePoll from "../components/Poll/createPoll";
import Nav from "./Nav";

class App extends Component {
	componentDidMount() {
		this.props.dispatch(handleInitialData());
	}

	render() {
		const polls = this.props.polls || {};

		return (
			<Router>
				<Fragment>
					{/* <LoadingBar /> */}
					<Nav />
					<Route path="/add" exact component={CreatePoll} />
					<Route
						path="/questions/:id"
						render={props =>
							polls[props.match.params.id] ? (
								<Poll poll={polls[props.match.params.id]} />
							) : null
						}
					/>
					<Route path="/" exact component={PollList} />
				</Fragment>
			</Router>
		);
	}
}

const mapStateToProps = state => {
	return {
		...state
	};
};

export default connect(mapStateToProps)(App);
