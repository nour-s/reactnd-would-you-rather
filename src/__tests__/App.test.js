import React from "react";
import { App } from "../components/App";
import { MemoryRouter } from "react-router";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

describe("App component", () => {
	it("renders without crashing", () => {
		const div = document.createElement("div");
		const props = {
			authedUser: { id: "nour" },
			users: {},
			polls: {}
		};

		const comp = <App {...props} />;
		const renderer = new ShallowRenderer();
		renderer.render(comp);
		const tree = renderer.getRenderOutput();
		expect(tree).toMatchSnapshot();
	});
});
