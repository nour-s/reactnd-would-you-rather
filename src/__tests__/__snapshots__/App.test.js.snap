// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without crashing 1`] = `
<BrowserRouter>
  <div
    className="app"
  >
    <header>
      <Nav />
    </header>
    <withRouter(Connect(PrivateRoute))
      component={[Function]}
      exact={true}
      path="/add"
    />
    <withRouter(Connect(PrivateRoute))
      path="/questions/:id"
      render={[Function]}
    />
    <withRouter(Connect(PrivateRoute))
      component={[Function]}
      exact={true}
      path="/leaderboard"
    />
    <withRouter(Connect(PrivateRoute))
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
  </div>
</BrowserRouter>
`;
